### PR TITLE
chore: add husky and commitlint for pre-commit linting

### LINKED ISSUE #126: Mention commitlint, standard, in readme, investigate husky
commitlint: in the action we just use the commitlint from [this GitHub Action](https://github.com/wagoid/commitlint-github-action).

It's worth trying to see if we can get Husky set up to automatically run these two checks by default for people when they get started so they're not caught off guard when checks on here don't pass.

_Originally posted by @simonv3 in https://github.com/resonatecoop/api/issues/125#issuecomment-1279964455_

### PR DESCRIPTION
**Pull request by simonv3**
Adding Husky to run commitlint and standard before commit to see if the code complies. 

Will catch people off guard less when they put in a pull request. 

Also added a postinstall script to have husky automatically install the hooks. I'm not sure this works if someone can test it out that would be appreciate it (just running `yarn install` after pulling down this branch).  

Closes #126

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -112,16 +113,21 @@
     "@babel/polyfill": "^7.8.3",
     "@babel/preset-env": "^7.8.4",
     "@babel/register": "^7.8.3",
+    "@commitlint/cli": "^17.1.2",
+    "@commitlint/config-conventional": "^17.1.0",
     "@faker-js/faker": "^7.5.0",
     "babel-plugin-add-module-exports": "^1.0.2",
     "chai": "^4.3.6",
     "dependency-check": "^4.1.0",
+    "husk": "^0.5.3",
```
Comment (fgblomqvist) [inline package.json:122]: Was this dependency intentional?

**Comment by simonv3** [comment]
@fgblomqvist possibly?

I was following the instructions here: https://typicode.github.io/husky/#/?id=yarn-2

Which are a bit ambiguous around when to enable and disable pinst: `and you're publishing it on a registry like npmjs`

**Comment by fgblomqvist** [comment]
Hmm yeah I think I understand. What they want to avoid is the `postinstall` running for people that are installing your package. However, this (`api`) is not a package that will be meant to be used as a dependency so it's safe if the postinstall always runs, which in turn means you don't need pinst at all.

**Comment by simonv3** [comment]
I'll see about removing `pinst` in a future PR. I just tested it out and I have to run the `yarn install` command outside of docker for it to succeed (so before `docker-compose up`) installing the husky hook.

**Pull request merged**